corelib-definitions/src/main/java/eu/europeana/corelib/definitions/model/facets/logic/CommonTagExtractor.java

```diff
@@ -0,0 +1,816 @@
+package eu.europeana.corelib.definitions.model.facets.logic;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Generates the common pure tags (mimetype and mediatype)
+ */
+public class CommonTagExtractor {
+
+    private static final Logger LOG = LogManager.getLogger(CommonTagExtractor.class.getName());
+
+    public static final Map<String, Integer> mimeTypes;
+
+    static {
+        final Map<String, Integer> mimeTypesDef = new HashMap<>();
```

Comment:
Performance wise it's better to always state a minimum size of you know a map is gonna be big. In this case you actually know the exact size you need, give that is the only parameter of the HashMap constructor.
.... = new HashMap<>(776);